convert/src/config.py

```diff
@@ -70,27 +78,49 @@ def get_request_session(*args: List[Any], **kwargs: Dict[str, Any]) -> Session:
 
 
 settings = Settings()
+logger_.info(f"{settings.s3_credentials_provider=}")
+
+
+class NotConfiguredException(Exception):
+    pass
+
+
+def create_boto3_config():
+    boto3_config = {}
+    if settings.s3_credentials_provider == "minio":
+        logger_.info(
+            f"S3_Credentials provider - {settings.s3_credentials_provider}"
+        )
+        boto3_config.update(
+            {
+                "aws_access_key_id": settings.minio_access_key,
+                "aws_secret_access_key": settings.minio_secret_key,
+                "endpoint_url": settings.minio_host,
+            }
+        )
+    elif settings.s3_credentials_provider == "aws_iam":
+        logger_.info(
+            f"S3_Credentials provider - {settings.s3_credentials_provider}"
+        )
+        # No additional updates to config needed - boto3 uses env vars
+    else:
+        raise NotConfiguredException(
+            "s3 connection is not properly configured - s3_credentials_provider is not set"
+        )
+    return boto3_config
 
 
 def get_minio_client() -> BaseClient:
     """Initialized s3 client by boto3 client"""
-    client = boto3.client(
-        "s3",
-        endpoint_url=settings.minio_host,
-        aws_access_key_id=settings.minio_access_key,
-        aws_secret_access_key=settings.minio_secret_key,
-    )
+    boto3_config = create_boto3_config()
+    client = boto3.client("s3", **boto3_config)
     return client
 
 
 def get_minio_resource() -> BaseClient:
     """Initialized s3 client by boto3 resource"""
-    client = boto3.resource(
-        "s3",
-        endpoint_url=settings.minio_host,
-        aws_access_key_id=settings.minio_access_key,
-        aws_secret_access_key=settings.minio_secret_key,
-    )
+    boto3_config = create_boto3_config()
+    client = boto3.resource("s3", **boto3_config)
```

Comment:
Let's remove one of the functions, they are identical

Reply:
they're not identical - 1st works with boto3.client(), 2nd - with boto3.resource()